--- src/screen.C
+++ src/screen.C
@@ -2506,7 +2506,10 @@
           /*
            * Actually do the drawing of the string here
            */
-          rxvt_font *font = (*fontset[GET_STYLE (rend)])[GET_FONT (rend)];
+          // I don't know why, but if GET_FONT(rend) returns a value greater than 15, a segmentation fault occurs.
+          // It can be reproduced when using :Rg of fzf-vim with 한글 font
+          auto font_index = GET_FONT(rend) >= 15 ? 3 : GET_FONT(rend);
+          rxvt_font *font = (*fontset[GET_STYLE (rend)])[font_index];
 
           if (ecb_likely (have_bg && back == Color_bg))
             {
